### ISSUE TITLE
[BUG]: multiple kv router might cause imbanlance result

### ISSUE THREAD

**Issue by spacewander** [ai-dynamo/dynamo#589]
### Describe the Bug

Sorry for using `[BUG]` template to discuss this potential risk.

The kv router's current schedule algorithm(https://github.com/ai-dynamo/dynamo/blob/main/lib/llm/src/kv_router/scheduler.rs) depends on the metrics from endpoints.

The scheduling algorithm selects the endpoint with the largest logits. Suppose we have multiple routers serving the same group of endpoints. Each router will choose the same best endpoint. Although a router modifies the endpoint's data after selection to make it less likely to be chosen again, this update is local to the individual machine and not perceivable by other routers.

Therefore, when multiple routers are in use, within a single metrics update cycle, the result may be imbalanced.

Example:
Suppose there are 16 routers. At a given moment, only one endpoint is the best choice.

Within a single metrics update cycle, every router forwards requests to this endpoint, causing it to instantly receive 16 requests. Each router, after selecting the endpoint, increments its own local counter by +1 (not +16), resulting in an underestimation of the actual request volume handled by the best endpoint.

### Steps to Reproduce

N/A

### Expected Behavior

N/A

### Actual Behavior

N/A

### Environment

N/A

**Comment by spacewander** [comment]
Maybe we can add some random in the algorithm to solve this.

**Comment by dzier** [comment]
Accepting this issues as a feature request.
